File: C#/IsSameTree.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int val=0, TreeNode left=null, TreeNode right=null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
public class Solution {
    public bool IsSameTree(TreeNode p, TreeNode q) {
        
        // Default
        if (p == null && q == null)
        {
            return true;
        }
        
        if (p != null && q != null)
        {
            //Console.WriteLine("P: " + p.val + ", Q: " + q.val);
        
            if (p.left != null && q.left != null)
            {
                if(IsSameTree(p.left, q.left) == false)
                {
                    return false;
                }
            }
            else if (p.left != null ^ q.left != null)
            {
                //Console.WriteLine("Not Symmetric");
                return false;
            }

            if (p.right != null && q.right != null)
            {
                if(IsSameTree(p.right, q.right) == false)
                {
                    return false;
                }
            }
            else if (p.right != null ^ q.right != null)
            {
                //Console.WriteLine("Not Symmetric");
                return false;   
            }

            if (p.val != q.val)
            {
                //Console.WriteLine("Not Equal");
                return false;
            }   
            
            return true;
        }
        
        return false;
        
    }
}

File: Python/isSameTree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def isSameTree(self, p: Optional[TreeNode], q: Optional[TreeNode]) -> bool:

        pNodeCheck = not(p)
        qNodeCheck = not(q)

        if (pNodeCheck ^ qNodeCheck):
            return False

        if (pNodeCheck & qNodeCheck):
            return True
        
        if not(p.val == q.val):
            return False

        leftCheck = self.isSameTree(p.left, q.left)
        rightCheck = self.isSameTree(p.right, q.right)

        return (leftCheck & rightCheck)

        